Softeq.XToolkit.PushNotifications.Droid/XFirebaseMessagingService.cs

```diff
@@ -4,28 +4,35 @@
 using System;
 using Android.App;
 using Firebase.Messaging;
+using Softeq.XToolkit.PushNotifications.Abstract;
+using Softeq.XToolkit.PushNotifications.Droid.Services;
+using Softeq.XToolkit.WhiteLabel;
 
 namespace Softeq.XToolkit.PushNotifications.Droid
 {
     [Service(Exported = true)]
     [IntentFilter(new[] { "com.google.firebase.MESSAGING_EVENT" })]
-    public class XFirebaseMessagingService : FirebaseMessagingService
+    internal class XFirebaseMessagingService : FirebaseMessagingService
     {
-        public static Action<string>? OnTokenRefreshed;
-        public static Action<RemoteMessage>? OnNotificationReceived;
+        private readonly IDroidFirebaseMessagingHandler _messagingHandler;
+
+        public XFirebaseMessagingService()
+        {
+            _messagingHandler = Dependencies.Container.Resolve<DroidPushNotificationsService>();
```

Comment:
are we sure that the container is ready for use at this time?

Reply:
Yes, the `Application` class where the container is initialised will be created before this class is created